rest_api/tests/api_test/post/test_rest_api_post.py

```diff
@@ -352,25 +354,57 @@ def test_api_post_batch_different_signer(self, setup):
                 LOGGER.info(data['error']['message'])
                 assert data['error']['code'] == 30
                 assert data['error']['title'] =='Submitted Batches Invalid' 
-                    
-    def test_api_post_batch_different_signer(self, setup):
+    
+    def test_rest_api_post_no_endpoint(self, setup):
+        
         signer_trans = get_signer() 
         intkey=create_intkey_transaction("set",[],50,signer_trans)
         translist=[intkey]
         signer_batch = get_signer()
         batch= create_batch(translist,signer_batch)
+        
         batch_list=[BatchList(batches=[batch]).SerializeToString()]
         for batc in batch_list:
             try:
-                response = post_batch(batc)
-                print(response)
-            except urllib.error.HTTPError as error:
-                LOGGER.info("Rest Api is not reachable")
-                data = json.loads(error.fp.read().decode('utf-8'))
-                LOGGER.info(data['error']['title'])
-                LOGGER.info(data['error']['message'])
-                assert data['error']['code'] == 30
-                assert data['error']['title'] =='Submitted Batches Invalid'
+                response = post_batch_no_endpoint(batc)
+            except urllib.error.HTTPError as e:
+                errdata = e.file.read().decode("utf-8")
+                errcode = e.code
+            assert errcode == 404
+            
+class TestPostMulTxns(RestApiBaseTest):
+     
+    def test_rest_api_post_valinv_txns(self, setup_batch_valinv_txns):
+        
+        initial_batch_length = setup_batch_valinv_txns['initial_batch_length']
+        expected_batch_length = setup_batch_valinv_txns['expected_batch_length']
+        initial_trn_length = setup_batch_valinv_txns['initial_trn_length']
+        expected_trn_length = setup_batch_valinv_txns['expected_trn_length']

```

Comment:
For post related tests , better create payload in fixtures and call here and then post from the test case